### DISCUSSION TITLE
Is it possible to override some drift query building logic ?

### DISCUSSION THREAD

**Discussion by Bill-Hetherington** [simolus3/drift#3067] [Q&A] [answered]
I made a MSSQL plugin to work with drift and it works quite well. But i have one issue around drift's query building, specifically the use of limit for example:
```
select(dsDao.menuItemMaster)
      ..where((tbl) => tbl.menuItemMasterID.equals(masterId))
      ..limit(1);
```
      
My plugin is meant to work for sqlExpress which does not support the "limit" clause instead using "Top". So as a workaround I could manipulate the query that drift builds using string manipulation prior to sending it off to my Db. 
But i'm interested to see if i could possibly override drift query building logic, as i did by making my own DatabaseDelegate, if possible without forking drift. The goal behind this plugin is to work seamlessly with drift so i would like any typical dao method like the example i provided to work with my plugin.
 Any insight would be appreciated.

**Comment by simolus3** [comment, marked as answer]
Drift doesn't allow patching the behavior of builtin clauses like `LIMIT` from outside of the package.

However, I'm happy to take contributions expanding drift support for different dialects. We define a dialect enum here. https://github.com/simolus3/drift/blob/06dbcc7ee70e2180648125581ea94f9aa1352c6f/drift/lib/src/runtime/query_builder/query_builder.dart#L95-L98

This could be expanded to include MSSQL-specific dialects, the implementation of the limit clause (`query_builder/components/limit.dart`) could then be changed to be aware of that option and generate the appropriate SQL.
I can't promise official support for that dialect without substantial tests, but even first and incomplete steps are welcome contributions and would make drift work with your use-case.
(also if your plugin is open source I'd love to reference it on the website for interested users)

**Reply by Bill-Hetherington** [reply]
Ok thank you, maybe in the near future you'll see a pull request for MSSQL dialect and a link to the plugin if i can open source it 😄
